ssl/ssl_locl.h

```diff
@@ -668,6 +668,8 @@ typedef struct raw_extension_st {
     int parsed;
     /* The type of this extension, i.e. a TLSEXT_TYPE_* value */
     unsigned int type;
+    /* The (0-based) index of this extension in its array, if present */
+    size_t idx;
```

Comment:
I found this confusing for two reasons. I wasn't sure what array we were talking about. Really you mean the position in the ClientHello. Secondly I was confusing the idx name with the TLSEXT_IDX_* values and thought they were somehow connected.